lib/spack/spack/main.py

```diff
@@ -868,7 +868,7 @@ def _main(argv=None):
         bootstrap_context = bootstrap.ensure_bootstrap_configuration()
 
     with bootstrap_context:
-        return finish_parse_and_run(parser, cmd_name, env_format_error)
+        finish_parse_and_run(parser, cmd_name, env_format_error)
```

Comment:
I'm temporarily reintroducing the bug and checking if the shell tests now fail

Reply:
Test failure is [here](https://github.com/spack/spack/runs/5359119769?check_suite_focus=true)